src/catalog/src/durable/impls/persist.rs

```diff
@@ -1079,25 +1099,47 @@ impl ReadOnlyDurableCatalogState for PersistCatalogState {
     ) -> Result<(Snapshot, Vec<VersionedEvent>, Vec<VersionedStorageUsage>), CatalogError> {
         self.sync_to_current_upper().await?;
         let snapshot = self.snapshot.clone();
-        let mut audit_events = Vec::new();
-        let mut storage_usages = Vec::new();
-        for StateUpdate { kind, ts: _, diff } in self.persist_snapshot().await {
-            soft_assert_eq_or_log!(1, diff, "updates should be consolidated");
-            match kind {
-                StateUpdateKind::AuditLog(audit_event, ()) => {
-                    let audit_event = AuditLogKey::from_proto(audit_event)?.event;
-                    audit_events.push(audit_event);
-                }
-                StateUpdateKind::StorageUsage(storage_usage, ()) => {
-                    let storage_usage = StorageUsageKey::from_proto(storage_usage)?.metric;
-                    storage_usages.push(storage_usage);
-                }
-                _ => {
-                    // Everything else is already cached in `self.snapshot`.
+        let (audit_events, storage_usage_events) = match (
+            self.audit_logs.cloned(),
+            self.storage_usage_events.cloned(),
+        ) {
+            (Some(audit_events), Some(storage_usage_events)) => {
+                (audit_events, storage_usage_events)
+            }
+            // We could check if each individual cache is populated, but this is unexpected and we
+            // need a full snapshot anyway. So we might as well compute both from a full snapshot.
+            _ => {
+                error!("audit events and storage usage events were not found in cache, so they were retrieved from persist, this is unexpected and bad for performance");
```

Comment:
Would something like `soft_assert_or_log` be good here so that this panics during dev and we can figure out why it's happening if it's unexpected?

Reply:
Good idea, just updated.

Reply:
Actually, I think I have to revert this. We do have some tests that test this code path, which is probably for the best. If we do end up accidentally running this in prod it'd be better if it was tested.